rio_tiler/mosaic/reader.py

```diff
@@ -76,15 +76,17 @@ def mosaic_reader(  # noqa: C901
             "'rio_tiler.mosaic.methods.base.MosaicMethodBase'"
         )
 
-    if not chunk_size:
-        chunk_size = threads if threads > 1 else len(mosaic_assets)
+    # if not chunk_size:
+    #    chunk_size = threads if threads > 1 else len(mosaic_assets)
```

Comment:
The only downside of Iterable is that we don't know its length but I'm fine not checking this for Iterable!